### ISSUE TITLE
Throw error on incorrect type in Image.putpixel

### ISSUE THREAD

**Issue by applemonkey496** [python-pillow/Pillow#4876]
*This text is pasted from my PR, #4875*

For `Image` objects with mode `L` (greyscale), when running `img.putpixel`, if the argument `xy` is not a tuple with types (int, int), the code raises a rather cryptic error:
```
SystemError: new style getargs format but argument is not a tuple
```
Again, if the argument **is** a tuple, but not with the types `(int, int)`, this error will come up, which is likely to confuse many people. I propose adding an `if` statement in the `putpixel` function to catch this case, and then to throw a clearer error message:
```
Expected argument xy to be a tuple (int, int) but got { xy }
```
Where `xy` is replaced by whatever acutal value of `xy` caused the error.

**See the [Pull Request](https://github.com/python-pillow/Pillow/pull/4875).**
____________
 * Related: [#2802](https://github.com/python-pillow/Pillow/issues/2802)
 * I got the idea to submit this pull request after reading [this answer](https://stackoverflow.com/a/43656642/11151112) on StackOverflow

**Comment by radarhere** [comment]
Hi. I'm not seeing this on my machine. Could you provide code so that we have an exact example, and let us know what version of Pillow you are using?

```python
import PIL
print(PIL.__version__)

from PIL import Image
im = Image.new("L", (1, 1))
im.putpixel("test", 0)
```
gives me
```
7.2.0
Traceback (most recent call last):
  File "example.py", line 6, in <module>
    im.putpixel("test", 0)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/PIL/Image.py", line 1744, in putpixel
    return self.im.putpixel(xy, value)
TypeError: argument 1 must be sequence of length 2, not 4
```

Trying with a tuple instead,
```python
import PIL
print(PIL.__version__)

from PIL import Image
im = Image.new("L", (1, 1))
im.putpixel(("test", "test"), 0)
```
I get
```
7.2.0
Traceback (most recent call last):
  File "example.py", line 6, in <module>
    im.putpixel(("test", "test"), 0)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/PIL/Image.py", line 1744, in putpixel
    return self.im.putpixel(xy, value)
TypeError: an integer is required (got type str)
```

**Comment by nulano** [comment]
Based on the SO answer, I found I can reproduce the given error message with `im.putpixel((0, 0), "hi")`. Looking at the code, the error is most likely coming from the line below, which should probably check the parameter is a tuple first. Is this the error you are trying to fix? If so, this should probably be done at the C level.
https://github.com/python-pillow/Pillow/blob/8deaebd5e0f1e00edfa02514a86fa3b52975a4f7/src/_imaging.c#L560

**Comment by applemonkey496** [comment]
@radarhere, I got the error because I was mapping my data from `(0,1)` to `(0,255)`, and so the output values were floats. So, when I ran:
```
for y in range(height):
  for x in range(width):
    img.putpixel((x,y), map[x,y])
```
My `map[x,y]` was a float, causing the error.

@nulano, I proposed catching the error in the Python code, because that's what I know, but yes, it might make more sense to correct it in the C code. If that's the case, someone else can add the fix, because my C knowledge is limited.
